rust/arrow/src/compute/kernels/aggregate.rs

```diff
@@ -152,9 +152,21 @@ where
             let data_chunks = data.chunks_exact(64);
             let remainder = data_chunks.remainder();
 
-            let bit_chunks = buffer.bit_chunks(array.offset(), array.len());
+            let buffer_slice = buffer.bit_slice().view(array.offset(), array.len());
+            let buffer_chunks = buffer_slice.chunks::<u64>();
+
+            let buffer_remainder_bits: u64 = buffer_chunks.remainder_bits();
+
+            let buffer_chunk_iter = buffer_chunks.interpret();
+
+            remainder.iter().enumerate().for_each(|(i, value)| {
```

Comment:
Can we move the remainder loop down again so that we are summing elements in the order that they are in the array?